### ISSUE TITLE
[BUG] Crashed vehicles drop child parts of parts, but parts will keep their own parts in their item

### ISSUE THREAD

**Issue by boot2big** [DonBruce64/MinecraftTransportSimulator#1560]
Example: Rims installed on a wheel, turbocharger installed on an engine, NPC on a seat, etc.
The vehicle will drop the rims from a wheel, turbocharger from an engine or the NPC from a seat, _as well_ as the part those child-parts were on... except that part itself has its own rim/turbo/NPC as if it wasn't dropped separately.

Ideally parts should prioritized being lobbed in with their parent parts rather than being separated, there's no point in forcing the user to put together modded engine just because they backed their front-engine car into a wall too hard. Likewise, if a tire is on a fancy rim, there's no practical reason to make the tire fall off in a crash, that's the whole point of having them pressurized and all! (Besides smoother rides and blablabla.)

tldr Plane crash dupe bug

**Comment by DonBruce64** [comment]
This poses issues.  Since I could just make each part drop as itself, but then you'd have invisible parts dropping.  I could ignore permanent parts, but then the save/load system would expect them during the next placement since they're normally only spawned during initial placement and saved with their parent.  Thoughts: just make the vehicle poof-ify entirely?

**Comment by boot2big** [comment]
I've had thoughts where we shouldn't make the vehicle poof _at all_ and that we could perhaps total the vehicle instead. Anything bigger than a semi truck (or if they're about the size of a semi truck, stronger than a plane) shouldn't just disappear as they do now anyways!
While it wouldn't hurt to ask other PAs on this I think that replacing vehicle poofing with force-totaling the vehicle would probably work better, especially if pack authors have their vehicles using destroyed models when the vehicle totals anyways.

**Comment by DonBruce64** [comment]
The problem with this assumption is most PAs don't do this.  They don't have totaled vehicle models, or even a repair recipe.  OCP, Vakshmat, Hero Aviation, and most other older packs weren't set up to support damage effects, much less totaled models.  I do not see instant-0-health being a good option in this case.

**Comment by boot2big** [comment]
I'd say it would be a good incentive for newer packs to start doing so then, plus most older packs do not even define their own health anyways and will have entire vehicles with just 100HP, which guns from newer packs always shred through within moments. (I _did_ try to suggest at least one method which would be basing a vehicle's HP off of its own emptyMass, whether directly or doubled!!)
Additionally I also feel that instant-0-healthing is a preferable alternative to any player over your-vehicle-go-poof-and-leaves-its-materials-or-parts, let alone your-vehicle-go-poof-and-you-getta-pound-sand. I'm not just thinking of this from a pack making perspective but also from a pack using perspective, particularly from the times _before_ I ever made any packs. It would allow for vehicles to become unusable when ramming them into the ground or a solid bedrock wall but would also be a lot less stressful than losing 100% of the crafting materials used into making an entire fighter jet. Or lets say, a whole freaking blimp! I'm not even sure if anyone actually has gone out of their way to craft the UNU blimp recipe but they would be very understandably pissed if **all** those resources just disappeared like that.

**Comment by DonBruce64** [comment]
Addressed in 22.8.0.
